addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -901,22 +901,25 @@ def activity_update(self):
                     count=allocation.number_of_days,
                     allocation_type=allocation.holiday_status_id.name
                 )
+                activity_type = self.env.ref(
+                    'hr_holidays.mail_act_leave_allocation_approval', raise_if_not_found=False
+                ) or self.env['mail.activity.type']
                 if allocation.state == 'confirm':
                     if allocation.holiday_status_id.responsible_ids:
                         user_ids = allocation.sudo()._get_responsible_for_approval().ids
                         for user_id in user_ids:
                             activity_vals.append({
-                                'activity_type_id': self.env.ref('hr_holidays.mail_act_leave_allocation_approval').id,
```

Comment:
Does it make sense to create them at all if it doesn't exist?

Doesn't it create a situation where they are not cleaned or "feedbacked" here below because they do not match the activity_type then?